### DISCUSSION TITLE
RP2350: Errata 9 Internal Pull-Down Latching issue

### DISCUSSION THREAD

**Discussion by Macrosii** [earlephilhower/arduino-pico#2373] [General]
Just wondering what the thoughts are on handling this?
Leave it to the user to handle for try to put something in the Core to help with this issue.

See [https://forums.raspberrypi.com/viewtopic.php?t=375631](url)

**Comment by Andy2No** [comment]
@Macrosii clicking that link didn't work, but repasting it does: https://forums.raspberrypi.com/viewtopic.php?t=375631

More of a worry is the issue by dhalbert, linked to from that thread:

https://github.com/raspberrypi/pico-feedback/issues/401 - a GPIO without an internal pull up set, pulled up externally via a 1M resistor latches to around 2V, and remains there if then pulled low with a resistor, needing a direct connection to ground to clear the condition (or presumably a sufficiently small value resistor).

Personally, I avoid the button directly to ground or 3.3V approach because if the pin is accidentally set as an output, e.g. by loading the wrong program/sketch which sets it to the opposite of what the button can connect it to, then pressing the button will draw too much current from the pin.  Perhaps there's current limiting to make that safe, in the Pico / Pico 2 but it seems like generally bad practice, so I don't do it - I connect a button through a 1.2k resistor, to limit the current to less than 3mA if that happens.

It would seem that the workaround for what @dhalbert describes (which it entirely reproducible) involves redesigning some circuits, at least, and quite likely having to redesign them in ways that are a bad habit to get into.

EDIT:  I missed seeing the post by @waiweng83 - apparently using a resistor of less than 9k to ground will unlatch it, after jumpering it to 3.3V.  I'm not clear if that means that a permanently connected resistor of less than 9k to ground would cope with a button shorting the pin to 3.3V.  Either way, it seems some redesigning of circuits might be needed, which previously worked with the RP2040 Pico, and we might need some guidance on how to wire a button to a Pico 2 reliably.

**Reply by earlephilhower** [reply]
#2380
